packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx

```diff
@@ -79,16 +130,163 @@ export const useDateRangePicker = ({
+                                    setInitialWeek(weekStart);
+                                    handleDateRangeChange([weekStart, weekEnd]);
+                                } else {
+                                    handleDateRangeChange([
+                                        initialWeek,
+                                        weekEnd,
+                                    ]);
+                                }
+                            }
+                            // Complete range selection
+                            else if (dates[0] && dates[1]) {
+                                const endWeek = getWeekBoundaries(dates[1]);
+                                if (initialWeek) {
+                                    handleDateRangeChange([
+                                        initialWeek,
+                                        endWeek.weekEnd,
+                                    ]);
+                                } else {
+                                    const startWeek = getWeekBoundaries(
+                                        dates[0],
+                                    );
+                                    handleDateRangeChange([
+                                        startWeek.weekStart,
+                                        endWeek.weekEnd,
+                                    ]);
+                                }
+                            }
+                        },
+                        numberOfColumns: 2,
+                        firstDayOfWeek: 0,
+                        hideOutsideDates: true,
+                        // Highlight the week of the selected date
+                        getDayProps: (date) => {
+                            const isSelected = Boolean(
+                                tempDateRange[0] &&
+                                    tempDateRange[1] &&
+                                    date >=
+                                        dayjs(tempDateRange[0])
+                                            .startOf('week')
+                                            .toDate() &&
+                                    date <=
+                                        dayjs(tempDateRange[1])
+                                            .endOf('week')
+                                            .toDate(),
+                            );
+                            return {
+                                selected: isSelected,
+                                inRange: isSelected,
+                                firstInRange: date.getDay() === 0 && isSelected, // Highlight the first day of the week
+                                lastInRange: date.getDay() === 6 && isSelected, // Highlight the last day of the week
+                            };
+                        },
+                    },
+                } satisfies WeekPickerConfig;
+            case TimeFrames.DAY:
+            default:
```

Comment:
should this be an exhaustive switch case? you can use `assertUnimplementedTimeframe` I think

Reply:
ooh like that! on it

Reply:
done: [refactor: use assertUnimplementedTimeframe](https://github.com/lightdash/lightdash/pull/12728/commits/3e261fcb8f338b3b7cb522588ef0b4c0a34ca928)